src/server/game/Spells/Spell.cpp

```diff
@@ -3318,7 +3318,7 @@ SpellCastResult Spell::prepare(SpellCastTargets const& targets, AuraEffect const
 
     // create and add update event for this spell
     _spellEvent = new SpellEvent(this);
-    m_caster->m_Events.AddEvent(_spellEvent, m_caster->m_Events.CalculateTime(1ms));
+    m_caster->m_Events.AddEvent(_spellEvent, m_caster->m_Events.CalculateTime(m_delay ? Milliseconds(m_delay) : 1ms));
```

Comment:
We could set m_delay to 1ms in the constructor and skip the 0 check here. what happens if m_delay is set to 0 anyway ?

Reply:
Not sure, but any CastSpell is scheduled for the next world tick.